File: generated-sources/javascript-closure-angular/mojang-api/API/Client/NameChange.js
goog.provide('API.Client.NameChange');

/**
 * A registered name change.
 * @record
 */
API.Client.NameChange = function() {}

/**
 * The new player name
 * @type {!string}
 * @export
 */
API.Client.NameChange.prototype.name;

/**
 * Time which the name was changed in UNIX Timestamp without milliseconds. Usually absent for the first entry.
 * @type {!number}
 * @export
 */
API.Client.NameChange.prototype.changedToAt;

File: generated-sources/scala-finch/mojang-api/src/main/scala/org/openapitools/models/NameChange.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import org.openapitools._

/**
 * A registered name change.
 * @param name The new player name
 * @param changedToAt Time which the name was changed in UNIX Timestamp without milliseconds. Usually absent for the first entry.
 */
case class NameChange(name: String,
                changedToAt: Option[Long]
                )

object NameChange {
    /**
     * Creates the codec for converting NameChange from and to JSON.
     */
    implicit val decoder: Decoder[NameChange] = deriveDecoder
    implicit val encoder: ObjectEncoder[NameChange] = deriveEncoder
}
